x-pack/plugins/security_solution/common/api/endpoint/actions/list_route.ts

```diff
@@ -34,6 +35,30 @@ export const EndpointActionListRequestSchema = {
         schema.string({ minLength: 1 }),
       ])
     ),
+    agentTypes: schema.maybe(
+      schema.oneOf([
+        schema.arrayOf(
+          schema.oneOf(
+            // @ts-expect-error TS2769: No overload matches this call
+            RESPONSE_ACTION_AGENT_TYPE.map((agentType) => schema.literal(agentType))
+          ),
+          {
+            minSize: 1,
+            validate: (agentTypes) => {
+              if (
+                !agentTypes.every((agentType) => RESPONSE_ACTION_AGENT_TYPE.includes(agentType))
+              ) {
+                return 'agentType must contain valid agent type value';
+              }
+            },
```

Comment:
`validate()` is not necessary I don't think. The `schema` already does value validations due to the use of `oneOf()`.

Reply:
c83b0677a7a6cd5fda0a20797651797b33797e7a